stable_baselines3/common/env_checker.py

```diff
@@ -11,11 +11,19 @@
 def _enforce_array_obs(observation_space: spaces.Space) -> bool:
     """
     Whether to check that the returned observation is a numpy array
-    it is not mandatory for `Dict` and `Tuple` spaces.
+    `Dict` and `Tuple` spaces are not guaranteed to contain only numpy arrays
     """
     return not isinstance(observation_space, (spaces.Dict, spaces.Tuple))
 
 
+def _enforce_array_act(action_space: spaces.Space) -> bool:
```

Comment:
I think this could be combined with the `_enforce_array_obs` by renaming the function to something common (e.g. `_ensure_array_spaces`), which is then used by both observation and action space.